src/AutoUI/Actions/Tags.tsx

```diff
@@ -31,12 +31,17 @@ export const Tags = <T extends AutoUIBaseResource<T>>({
 
 	const { data: items, isLoading } = useRequest(
 		async () => {
-			if (checkedState === 'all' && sdk?.tags && 'getAll' in sdk.tags) {
+			if (
+				!selected &&
+				checkedState === 'all' &&
+				sdk?.tags &&
+				'getAll' in sdk.tags
+			) {
 				return await sdk.tags.getAll(internalPineFilter);
 			}
 			return selected;
 		},
-		[internalPineFilter, checkedState, sdk],
+		[internalPineFilter, checkedState, sdk, selected],
```

Comment:
```suggestion
[internalPineFilter, checkedState, sdk?.tags?.getAll, selected == null],
```

but probably better to do a `const getAllTags = sdk?.tags && 'getAll' in sdk.tags ? sdk.tags.getAll : null;` outside